phpBB/assets/javascript/installer.js

```diff
@@ -0,0 +1,400 @@
+/**
+ * Installer's AJAX frontend handler
+ */
+
+(function($) { // Avoid conflicts with other libraries
+	'use strict';
+
+	// Installer variables
+	var pollTimer = null;
+	var nextReadPosition = 0;
+	var progressBarTriggered = false;
+	var progressTimer = null;
+	var currentProgress = 0;
+
+	// Template related variables
+	var $contentWrapper = $('.install-body').find('.main');
+
+	// Intercept form submits
+	interceptFormSubmit($('#install_install'));
+
+	/**
+	 * Creates an XHR object
+	 *
+	 * jQuery cannot be used as the response is streamed, and
+	 * as of now, jQuery does not provide access to the response until
+	 * the connection is not closed.
+	 *
+	 * @return XMLHttpRequest
+	 */
+	function createXhrObject() {
+		return new XMLHttpRequest();
+	}
+
+	/**
+	 * Displays error, warning and log messages
+	 *
+	 * @param type
+	 * @param messages
+	 */
+	function addMessage(type, messages) {
+		// Get message containers
+		var $errorContainer = $('#error-container');
+		var $warningContainer = $('#warning-container');
+		var $logContainer = $('#log-container');
+
+		var $title, $description, $msgElement, arraySize = messages.length;
+		for (var i = 0; i < arraySize; i++) {
+			$msgElement = $('<div />');
+			$title = $(document.createElement('strong'));
+			$title.text(messages[i].title);
+			$msgElement.append($title);
+
+			if (messages[i].hasOwnProperty('description')) {
+				$description = $(document.createElement('p'));
```

Comment:
Any reason why you are not using the same as here? https://github.com/phpbb/phpbb/pull/3514/files#diff-7bf94728366cca27f7f0afb9e259924bR147